source/application/AMPIC887DriverServer/AMPIC887ConsoleApplication.cpp

```diff
@@ -70,23 +66,507 @@ void AMPIC887ConsoleApplication::onActiveControllerChangeIssued(const QString &c
 	}
 }
 
-void AMPIC887ConsoleApplication::onGCS2CommandIssued(AMGCS2Command* command)
+void AMPIC887ConsoleApplication::onControllerOutputReceived(const QString &outputString)
 {
-	if(!controllerCollection_.activeController()) {
+	consoleInputHandler_->writeLineToStandardOutput(outputString);
+}
+
+void AMPIC887ConsoleApplication::onControllerErrorEncountered(const QString &errorMessage)
+{
+	consoleInputHandler_->writeLineToStandardError(errorMessage);
+}
+
+void AMPIC887ConsoleApplication::onSetCommandLevelCommandIssued(AMGCS2::ControllerCommandLevel commandLevel,
+																const QString &password)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
```

Comment:
That would also definitely be okay.  Feel free to make those changes (... so many changes...)